lib/Net/Fastly/Client.pm

```diff
@@ -70,6 +70,9 @@ sub new {
 
     # If we're fully authed (i.e username and password ) then we need to log in
     my $res = $self->_ua->_post('/login', {}, user => $self->{user}, password => $self->{password});
+
+    warn "DEPRECATION WARNING: Username/password authentication is deprecated and will not be available starting September 2020; please migrate to API tokens as soon as possible.";
```

Comment:
I verified that you can pass username/password into the constructor:
```
✅ doublethink% perl -Ilib -MNet::Fastly::Client -E 'Net::Fastly::Client->new( username => "foo", password => "bar" )'
DEPRECATION WARNING: Username/password authentication is deprecated and will not be available starting September 2020; please migrate to API tokens as soon as possible. at lib/Net/Fastly/Client.pm line 74.
Unauthorized at lib/Net/Fastly/Client.pm line 78.
```

I think therefore using `carp` would be good here, since that will point to the calling location rather than inside the constructor.  `carp` works just like `warn`, except that it is implemented in a module so we need to import it from the `Carp` module (https://metacpan.org/pod/Carp).  So the top of the module looks something like:

```perl
package Net::Fastly::Client;

use strict;
use warnings;
use Carp qw( carp );
use JSON::XS;
```

and this line becomes:

```suggestion
    carp "DEPRECATION WARNING: Username/password authentication is deprecated and will not be available starting September 2020; please migrate to API tokens as soon as possible.";
```

Reply:
👍 Excellent, added carp